src/test/scala/it/pagopa/interop/purposeprocess/PurposeApiServiceSpec.scala

```diff
@@ -1512,6 +1513,192 @@ class PurposeApiServiceSpec extends AnyWordSpecLike with SpecHelper with Scalate
+      val purposeVersionId2 = UUID.randomUUID()
+      val purposeVersionId3 = UUID.randomUUID()
+      val eserviceId        = UUID.randomUUID()
+      val descriptorId      = UUID.randomUUID()
+
+      implicit val context: Seq[(String, String)] =
+        Seq("bearer" -> bearerToken, USER_ROLES -> "admin", ORGANIZATION_ID_CLAIM -> consumerId.toString)
+
+      val version2_1 = SpecData.purposeVersion.copy(id = UUID.randomUUID(), state = Active, dailyCalls = 1000)
+      val purpose2 = SpecData.purpose.copy(eserviceId = eserviceId, consumerId = consumerId, versions = Seq(version2_1))
+
+      val path: String                               = "/here/there/foo/bar.pdf"
+      val document: PersistentPurposeVersionDocument =
+        PersistentPurposeVersionDocument(documentId, "application/pdf", path, SpecData.timestamp)
+
+      val version1_1 = SpecData.purposeVersion.copy(id = purposeVersionId1, state = Active)
+      val version1_2 = SpecData.purposeVersion.copy(
+        id = purposeVersionId2,
+        state = WaitingForApproval,
+        riskAnalysis = Some(document),
+        firstActivationAt = Some(SpecData.timestamp)
+      )
+
+      val purpose =
+        SpecData.purpose.copy(
+          id = purposeId,
+          versions = Seq(version1_1, version1_2),
+          consumerId = consumerId,
+          eserviceId = eserviceId
+        )
+
+      val purposes                 = Seq(purpose, purpose2)
+      val seed: PurposeVersionSeed = PurposeVersionSeed(dailyCalls = 1000)
+
+      val purposeVersion = PersistentPurposeVersion(
+        id = purposeVersionId3,
+        state = WaitingForApproval,
+        createdAt = SpecData.timestamp,
+        updatedAt = None,
+        expectedApprovalDate = None,
+        dailyCalls = seed.dailyCalls,
+        riskAnalysis = Some(document),
+        firstActivationAt = Some(SpecData.timestamp),
+        suspendedAt = None
+      )
+
+      mockPurposeRetrieve(purposeId, purpose)
+
+      (mockPurposeManagementService
+        .createPurposeVersion(_: UUID, _: PurposeManagementDependency.PurposeVersionSeed)(_: Seq[(String, String)]))
+        .expects(purposeId, seed.toManagement, context)
+        .once()
+        .returns(Future.successful(SpecData.dependencyPurposeVersion.copy(id = purposeVersionId3)))
+
+      val descriptor = SpecData.descriptor.copy(id = descriptorId, dailyCallsPerConsumer = 100)
+      val eService   = SpecData.eService.copy(id = eserviceId, descriptors = Seq(descriptor))
+
+      mockEServiceRetrieve(purpose.eserviceId, eService)
+
+      mockVersionLoadValidation(purpose, purposes, descriptorId)
+
+      val updatedVersion =
+        SpecData.dependencyPurposeVersion.copy(
+          id = purposeVersionId3,
+          state = PurposeManagementDependency.PurposeVersionState.WAITING_FOR_APPROVAL,
+          riskAnalysis = Some(
+            PurposeManagementDependency.PurposeVersionDocument(documentId, "application/pdf", path, SpecData.timestamp)
+          ),
```

Comment:
just a minor mock fix: the risk analysis document is created only when the version state is changed to ACTIVE for the first time.
In this case the document is `None`
```suggestion
          riskAnalysis = None,
```